File: rest-service/src/main/java/mb/api/service/valid/SequenceValidationError.java
package mb.api.service.valid;

public class SequenceValidationError
{
  private final int sequence;
  private final int  line;
  private final int  position;
  private final char character;

  public SequenceValidationError(int sequence, int line, int position, char character) {
    this.sequence  = sequence;
    this.line      = line;
    this.position  = position;
    this.character = character;
  }

  public int getSequence() {
    return sequence;
  }

  public int getLine() {
    return line;
  }

  public int getPosition() {
    return position;
  }

  public char getCharacter() {
    return character;
  }
}

File: rest-service/src/main/java/mb/api/service/valid/SequenceValidationError.kt
package mb.api.service.valid

sealed interface SequenceValidationError {
  val message: String
}

class SequenceCharacterValidationError(
  val sequence: Int,
  val line: Int,
  val position: Int,
  val character: Char
) : SequenceValidationError {
  override val message
    get() = "Invalid character '$character' in sequence $sequence at line $line, position $position."
}

class SequenceLengthValidationError(
  val length:    Int,
  val maxLength: Int
) : SequenceValidationError {
  override val message
    get() = "Sequence too long, max length is $maxLength input length was $length."
}